frontend/src/app/testQueue/TestCardForm/diseaseSpecificComponents/TestResultInputGroup.tsx

```diff
@@ -0,0 +1,123 @@
+import { Alert } from "@trussworks/react-uswds";
+
+import {
+  MultiplexResultInput,
+  SupportedDiseaseTestPerformed,
+} from "../../../../generated/graphql";
+import { DevicesMap } from "../../QueueItem";
+import RadioGroup, {
+  RadioGroupOptions,
+} from "../../../commonComponents/RadioGroup";
+import { TEST_RESULTS } from "../../../testResults/constants";
+import { TEST_RESULT_DESCRIPTIONS } from "../../../constants";
+
+interface Props {
+  testOrderId: string;
+  testResults: MultiplexResultInput[];
+  deviceId: string;
+  devicesMap: DevicesMap;
+  onChange: (value: MultiplexResultInput[]) => void;
+}
+
+type TestResultRecord = Record<string, TEST_RESULTS>;
+
+const convertTestResultRecordToArray = (testResultRecord: TestResultRecord) => {
+  return Object.entries(testResultRecord).map((entry): MultiplexResultInput => {
+    return {
+      diseaseName: entry[0],
+      testResult: entry[1],
+    };
+  });
+};
+
+const convertMultiplexResultArrayToRecord = (
+  testResults: MultiplexResultInput[]
+) => {
+  const testResultRecords: TestResultRecord = {};
+  testResults.forEach((value) => {
+    if (value.diseaseName) {
+      testResultRecords[value.diseaseName] =
+        (value.testResult as TEST_RESULTS) ?? TEST_RESULTS.UNKNOWN;
+    }
+  });
+  return testResultRecords;
+};
+
+export const TestResultInputGroup = ({
+  testOrderId,
+  testResults,
+  deviceId,
+  devicesMap,
+  onChange,
+}: Props) => {
+  const supportedDiseaseTests =
+    devicesMap.get(deviceId)?.supportedDiseaseTestPerformed ?? [];
+
+  if (supportedDiseaseTests.length === 0) {
+    return (
+      <Alert headingLevel={"h3"} type={"error"}>
+        This device doesn't have any supported disease tests configured. Please
+        ask your organization admin to add the correct test.
```

Comment:
Checking with design and content to see how we want to handle this error case where the device doesn't have any diseases configured, but this seems to be an edge case only encountered on some of the dev environments that still have pre-multiplex devices in the database